test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -954,4 +963,53 @@ object Assertion extends AssertionVariants {
         .withCode("isJustInterrupted")
     )
 
+  sealed trait Arguments
+  object Arguments {
+    /*
+     * Creates a string representation of a class name.
+     */
+    def className[A](implicit C: ClassTag[A]): String =
+      try {
+        C.runtimeClass.getSimpleName
+      } catch {
+        // See https://github.com/scala/bug/issues/2034.
+        case t: InternalError if t.getMessage == "Malformed class name" =>
+          C.runtimeClass.getName
+      }
+
+    /**
+     * Creates a string representation of a field accessor.
+     */
+    def field(name: String): String =
+      "_." + name
+
+    def mapKey[V](v: V): String =
+      v match {
+        case assertion: Assertion[_] =>
+          assertion.render
+        case s => s.toString
+
+      }
+
+    /**
+     * Construct a `RenderParam` from an `Assertion`.
+     */
+    def param[A](assertion: Assertion[A]): Arguments =
```

Comment:
the zio styleguide might get published sometimes, but one common rule is to avoid overloading in favour of constructor with specific names. How about:
```
def value[a]: (a, Option[String] = None) => Arguments
def assertion[a]: Assertion[a] => Arguments
def `type`[a]: ClassTag[a] ?=> Arguments
```

Reply:
@regiskuckaertz  updated. I've also moved all the rendering to the toString methods of the Argument types. not sure if that's better or worse?